modules/aws/s3.go

```diff
@@ -260,6 +263,81 @@ func EmptyS3BucketE(t testing.TestingT, region string, name string) error {
 	return err
 }
 
+// PutS3BucketLogging enables logging for the given S3 bucket in the given region
+// Make sure that the logsTargetBucket exists prior to using it here.
+func PutS3BucketLogging(t testing.TestingT, region string, bucketName string, logsTargetBucketName string) {
+	err := PutS3BucketLoggingE(t, region, bucketName, logsTargetBucketName)
+	require.NoError(t, err)
+}
+
+// PutS3BucketLoggingE enables logging for the given S3 bucket in the given region
+// Make sure that the logsTargetBucket exists prior to using it here.
+func PutS3BucketLoggingE(t testing.TestingT, region string, sourceBucketName string, logsTargetBucketName string) error {
+	logger.Logf(t, "Applying logging to S3 bucket %s in %s", sourceBucketName, region)
+	logger.Logf(t, "Setting TargetBucket to be %s", logsTargetBucketName)
+
+	s3Client, err := NewS3ClientE(t, region)
+	if err != nil {
+		return err
+	}
+
+	logsUri := fmt.Sprintf("uri=%s", logDeliveryUri)
+
+	//Put ACL permissions on the targetBucket, as it's different from the origin S3 bucket
+	aclInput := s3.PutBucketAclInput{
+		Bucket:       aws.String(logsTargetBucketName),
+		GrantWrite:   aws.String(logsUri),
+		GrantReadACP: aws.String(logsUri),
+	}
+
+	if _, err := s3Client.PutBucketAcl(&aclInput); err != nil {
+		return err
+	}
+
+	//enable logging on the origin S3 bucket
+	params := &s3.PutBucketLoggingInput{
+		Bucket: aws.String(sourceBucketName),
+		BucketLoggingStatus: &s3.BucketLoggingStatus{
+			LoggingEnabled: &s3.LoggingEnabled{
+				TargetBucket: aws.String(logsTargetBucketName),
+				TargetPrefix: aws.String("/"),
+			},
+		},
+	}
+
+	_, err = s3Client.PutBucketLogging(params)
+	return err
+}
+
+// GetS3BucketLoggingTarget fetches the given bucket's logging configuration status and returns it as a string
+func GetS3BucketLoggingTarget(t testing.TestingT, awsRegion string, bucket string) string {
+	loggingTarget, err := GetS3BucketLoggingTargetE(t, awsRegion, bucket)
+	require.NoError(t, err)
+
+	return loggingTarget
+}
+
+// GetS3BucketLoggingE fetches the given bucket's versioning configuration status and returns it as a string
+func GetS3BucketLoggingTargetE(t testing.TestingT, awsRegion string, bucket string) (string, error) {
+	s3Client, err := NewS3ClientE(t, awsRegion)
+	if err != nil {
+		return "", err
+	}
+
+	res, err := s3Client.GetBucketLogging(&s3.GetBucketLoggingInput{
+		Bucket: &bucket,
+	})
+	if err != nil {
+		return "", err
+	}
+
+	if res.LoggingEnabled != nil {
+		return aws.StringValue(res.LoggingEnabled.TargetBucket), nil
+	}
+
+	return "", err
+}
```

Comment:
I'm a bit on the fence about this one, given that it's a thin wrapper around the API, but since the s3 based functionality isn't really complete ([completeness litmus test](https://github.com/gruntwork-io/terratest/issues/421#issuecomment-567350190)) without looking for access logging, I think it is ok to include this getter function.

Reply:
Just checking if I've understood you correctly - is it that the function is a bit too specific?

I guess a bit of context would help - [this is actually going to help me test using `terratest` the new `terragrunt` `accesslogging_bucket_name`. ](https://github.com/gruntwork-io/terragrunt/pull/1400#pullrequestreview-518125751)

However, after you've pointed me to the `litmus test`, I can clearly see this is only needed as functionality in that one specific scenario.

Reply:
The specific litmus test I am on the fence about is the "complexity" test, which states:

> Complexity: we ask that you only contribute infrastructure and validation helpers for code that is relatively complex to do from scratch. For example, a helper that merely wraps an existing function in the AWS or GCP SDK is not a great choice, as the wrapper isn’t contributing much value, but is bloating the Terratest API. On the other hand, helpers that expose simple APIs for complex logic are great contributions: ssh.CheckSshCommand is a great example of this, as it provides a simple one-line interface for dozens of lines of complicated SSH logic.

This function is a thin wrapper around the AWS SDK, which is why it can be considered as bloating the Terratest API. For example, in your test code, you can do something like the following:

```go
s3Client := NewS3Client(t, awsRegion)
res, err := GetBucketLogging(&s3.GetBucketLoggingInput{Bucket: &bucket})
require.NoError(t, err)
// Assert that logging is enabled
require.NotNil(t, res.LoggingEnabled)
// Assert that logging is done to the expected target bucket
assert.Equal(t, expectedTargetBucket, aws.StringValue(res.LoggingEnabled.TargetBucket))
```

This code is readable, and more direct in the assertions. Of course, arguably one line is better than five lines, and it will be subjective which is better (having a terratest function that simplifies these 5 lines of code, or writing them inline). Moreover, access log checks is very common, so one could argue that the terratest API is incomplete without it. This is why I am on the fence, but leaning towards including it.

Reply:
OK, this is really insightful, Yori! Thanks!!

I've spent some time adding this as a terraform test instead. I've got a bit more cleanup to do, so will move to draft for now, and then move it back into "Ready for Review" as I work through stuff (most likely on Monday).